### PR TITLE
[grafana] Create role rules on notifier toggle

### PR DESCRIPTION
**Pull request by garettmd**
I noticed that when using _only_ the notifiers sidecar (not alongside the dashboards or datasources sidecars), the rules necessary for both Role and ClusterRole are not created. I don't know if this is intentional or just not used very often, but in my case it was preventing me from deploying Grafana. Adding another bit `or` logic resolves this.

### CODE REVIEW

File: charts/grafana/templates/clusterrole.yaml [hunk 1]
```diff
@@ -9,9 +9,9 @@ metadata:
 {{ toYaml . | indent 4 }}
 {{- end }}
   name: {{ template "grafana.fullname" . }}-clusterrole
-{{- if or .Values.sidecar.dashboards.enabled (or .Values.sidecar.datasources.enabled .Values.rbac.extraClusterRoleRules) }}
+{{- if or .Values.sidecar.dashboards.enabled (or .Values.sidecar.datasources.enabled (or .Values.sidecar.notifiers.enabled .Values.rbac.extraClusterRoleRules)) }}
```
Comment (zanhsieh) [inline charts/grafana/templates/clusterrole.yaml:12]: Why nested here? Just put:
```
{{- if or .Values.sidecar.dashboards.enabled .Values.sidecar.datasources.enabled .Values.sidecar.notifiers.enabled .Values.rbac.extraClusterRoleRules }}
```

File: charts/grafana/templates/clusterrole.yaml [hunk 2]
```diff
@@ -9,9 +9,9 @@ metadata:
...
 rules:
-{{- if or .Values.sidecar.dashboards.enabled .Values.sidecar.datasources.enabled }}
+{{- if or .Values.sidecar.dashboards.enabled (or .Values.sidecar.notifiers.enabled .Values.sidecar.datasources.enabled )}}
```
Comment (zanhsieh) [inline charts/grafana/templates/clusterrole.yaml:14]: Just put
```
{{- if or .Values.sidecar.dashboards.enabled .Values.sidecar.notifiers.enabled .Values.sidecar.datasources.enabled }}
```

File: charts/grafana/templates/role.yaml [hunk 3]
```diff
@@ -10,15 +10,15 @@ metadata:
   annotations:
 {{ toYaml . | indent 4 }}
 {{- end }}
-{{- if or .Values.rbac.pspEnabled (and .Values.rbac.namespaced (or .Values.sidecar.dashboards.enabled (or .Values.sidecar.datasources.enabled .Values.rbac.extraRoleRules))) }}
+{{- if or .Values.rbac.pspEnabled (and .Values.rbac.namespaced (or .Values.sidecar.dashboards.enabled (or .Values.sidecar.datasources.enabled (or .Values.sidecar.notifiers.enabled .Values.rbac.extraRoleRules)))) }}
```
Comment (zanhsieh) [inline charts/grafana/templates/role.yaml:13]: Why not simplify as:
```
{{- if or .Values.rbac.pspEnabled (and .Values.rbac.namespaced (or .Values.sidecar.dashboards.enabled  .Values.sidecar.datasources.enabled  .Values.sidecar.notifiers.enabled .Values.rbac.extraRoleRules)) }}
```

File: charts/grafana/templates/role.yaml [hunk 4]
```diff
@@ -10,15 +10,15 @@ metadata:
...
 rules:
 {{- if .Values.rbac.pspEnabled }}
 - apiGroups:      ['extensions']
   resources:      ['podsecuritypolicies']
   verbs:          ['use']
   resourceNames:  [{{ template "grafana.fullname" . }}]
 {{- end }}
-{{- if and .Values.rbac.namespaced (or .Values.sidecar.dashboards.enabled .Values.sidecar.datasources.enabled) }}
+{{- if and .Values.rbac.namespaced (or .Values.sidecar.dashboards.enabled (or .Values.sidecar.notifiers.enabled .Values.sidecar.datasources.enabled)) }}
```
Comment (zanhsieh) [inline charts/grafana/templates/role.yaml:21]: Why not simplify as:
```{{- if and .Values.rbac.namespaced (or .Values.sidecar.dashboards.enabled .Values.sidecar.notifiers.enabled .Values.sidecar.datasources.enabled) }}
```

**Comment by zanhsieh** [comment]
@garettmd 
In case you need some proof / evaluation, here is a Go playground:

https://go.dev/play/p/HM44WvVrEEF

**Comment by garettmd** [comment]
Thanks for the input @zanhsieh. I was just following the nested approach that already exists in the code. I'll add your proposed changes.

File: charts/grafana/templates/role.yaml (hunk 4 above)
Comment (zanhsieh) [reply charts/grafana/templates/role.yaml:21]:
```suggestion
{{- if and .Values.rbac.namespaced (or .Values.sidecar.dashboards.enabled .Values.sidecar.notifiers.enabled .Values.sidecar.datasources.enabled) }}
```

File: charts/grafana/templates/role.yaml (hunk 3 above)
Comment (zanhsieh) [reply charts/grafana/templates/role.yaml:13]:
```suggestion
{{- if or .Values.rbac.pspEnabled (and .Values.rbac.namespaced (or .Values.sidecar.dashboards.enabled  .Values.sidecar.datasources.enabled  .Values.sidecar.notifiers.enabled .Values.rbac.extraRoleRules)) }}
```

File: charts/grafana/templates/clusterrole.yaml (hunk 2 above)
Comment (zanhsieh) [reply charts/grafana/templates/clusterrole.yaml:14]:
```suggestion
{{- if or .Values.sidecar.dashboards.enabled .Values.sidecar.notifiers.enabled .Values.sidecar.datasources.enabled }}
```

File: charts/grafana/templates/clusterrole.yaml (hunk 1 above)
Comment (zanhsieh) [reply charts/grafana/templates/clusterrole.yaml:12]:
```suggestion
{{- if or .Values.sidecar.dashboards.enabled .Values.sidecar.datasources.enabled .Values.sidecar.notifiers.enabled .Values.rbac.extraClusterRoleRules }}
```

**Pull request closed without merging**